zhaquirks/bosch/rbsh_trv0_zb_eu.py

```diff
@@ -0,0 +1,454 @@
+WINDOW_OPEN_ATTR_ID = 0x4042
+
+# Boost preset mode.
+BOOST_ATTR_ID = 0x4043
+
+"""Bosch specific user interface attribute ids."""
+
+# Display orientation with values BoschDisplayOrientation.
+SCREEN_ORIENTATION_ATTR_ID = 0x400B
+
+# Displayed temperature with values BoschDisplayedTemperature.
+DISPLAY_MODE_ATTR_ID = 0x4039
+
+# Display on-time (5s - 30s).
+SCREEN_TIMEOUT_ATTR_ID = 0x403A
+
+# Display brightness (0 - 10).
+SCREEN_BRIGHTNESS_ATTR_ID = 0x403B
+
+# Control sequence of operation (heating/cooling)
+CTRL_SEQUENCE_OF_OPERATION_ID = 0x001B
+
+class BoschOperatingMode(t.enum8):
+    """Bosh operating mode attribute values."""
+
+    Schedule = 0x00
+    Manual = 0x01
+    Pause = 0x05
+
+
+class State(t.enum8):
+    """Binary attribute (window open) value."""
+
+    Off = 0x00
+    On = 0x01
+
+
+class BoschDisplayOrientation(t.enum8):
+    """Bosch display orientation attribute values."""
+
+    Normal = 0x00
+    Flipped = 0x01
+
+
+class BoschDisplayedTemperature(t.enum8):
+    """Bosch displayed temperature attribute values."""
+
+    Target = 0x00
+    Measured = 0x01
+
+
+class BoschControlSequenceOfOperation(t.enum8):
+    """Supported ControlSequenceOfOperation modes."""
+
+    Cooling = ControlSequenceOfOperation.Cooling_Only
+    Heating = ControlSequenceOfOperation.Heating_Only
+
+
+"""HA thermostat attribute that needs special handling in the Bosch thermostat entity."""
+SYSTEM_MODE_ATTR = Thermostat.AttributeDefs.system_mode
+
+"""Bosch operating mode to HA system mode mapping."""
+OPERATING_MODE_TO_SYSTEM_MODE_MAP = {
+    BoschOperatingMode.Schedule: Thermostat.SystemMode.Auto,
+    BoschOperatingMode.Manual: Thermostat.SystemMode.Heat,
+    BoschOperatingMode.Pause: Thermostat.SystemMode.Off,
+    "BoschOperatingMode.Schedule": Thermostat.SystemMode.Auto,
+    "BoschOperatingMode.Manual": Thermostat.SystemMode.Heat,
+    "BoschOperatingMode.Pause": Thermostat.SystemMode.Off,
+}
+
+"""HA system mode to Bosch operating mode mapping."""
+SYSTEM_MODE_TO_OPERATING_MODE_MAP = {
+    Thermostat.SystemMode.Off: BoschOperatingMode.Pause,
+    Thermostat.SystemMode.Heat: BoschOperatingMode.Manual,
+    Thermostat.SystemMode.Cool: BoschOperatingMode.Manual,
+    Thermostat.SystemMode.Auto: BoschOperatingMode.Schedule,
+    "SystemMode.Off": BoschOperatingMode.Pause,
+    "SystemMode.Heat": BoschOperatingMode.Manual,
+    "SystemMode.Cool": BoschOperatingMode.Manual,
+    "SystemMode.Auto": BoschOperatingMode.Schedule,
+}
+
+DISPLAY_ORIENTATION_ENUM_TO_INT_MAP = {
+    0x00: 0x00,
+    0x01: 0x01,
+    "BoschDisplayOrientation.Normal": 0x00,
+    "BoschDisplayOrientation.Flipped": 0x01,
+}
+
+
+class BoschThermostatCluster(CustomCluster, Thermostat):
+    """Bosch thermostat cluster."""
+
+    class AttributeDefs(Thermostat.AttributeDefs):
+        """Bosch thermostat manufacturer specific attributes."""
+
+        operating_mode = ZCLAttributeDef(
+            id=t.uint16_t(OPERATING_MODE_ATTR_ID),
```

Comment:
You shouldn't need to wrap this in an `t.uint16_t`.
(same for all other `ZCLAttributeDef` too)

Reply:
Removed.